[doc] Improve joda-time migration guide (#60499)
elastic/elasticsearch: docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

formatting, and calculations. This guide is designed to help you determine
if your cluster is impacted and, if so, prepare for the upgrade.

*You do not need to convert joda-time date formats on indices created in {es} 6.8
before upgrading to {es} 7.7 or newer*

However you will have to make sure mappings for newly created indices are using
java-time patterns.
Note: {es} 7.0-7.6 does not work with joda-time patterns. It was fixed in {es} 7.7
see: {es-pull}52555[#52555]

You need to update your pipelines or templates, because creating new indices with
joda-time patterns in mappings is not allowed.
<<java-time-migration-update-ingest-pipelines>>
<<java-time-migration-update-index-templates>>

[discrete]
[[java-time-convert-date-formats]]
=== Convert date formats

> You can start using java-time patterns in {es} 6.8 by prefixing java-time
> date formats with an `8`.

Reviewer: ```suggestion
To use java time in 6.8, prefix the date format with an `8`.
```

[[java-time-migration-incompatible-date-formats]]
==== Incompatible date formats
Custom date formats containing the following joda-time literals should be

> migrated.

Reviewer: ```suggestion
converted to their java-time equivalents.
```
